_posts/2013-01-11-how-to-hide-content.md

```diff
@@ -9,33 +9,39 @@ categories:
-
-There are also real world situations where you want to hide elements (e.g., a `form label`), but you still want text to be read by a screen reader. The "clip pattern" will hide the content visually, yet provide the content to screen readers[^clip]. Unlike positioning and negative text-indent, it works with rtl languages for localization.
-
-    .visually-hidden { /*https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html*/
+    .visually-hidden { /* https://snook.ca/archives/html_and_css/hiding-content-for-accessibility */
         position: absolute !important;
+        height: 1px; width: 1px; 
+        overflow: hidden;
         clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
         clip: rect(1px, 1px, 1px, 1px);
-        padding:0 !important;
-        border:0 !important;
-        height: 1px !important;
-        width: 1px !important;
-        overflow: hidden;
     }
-    body:hover .visually-hidden a, body:hover .visually-hidden input, body:hover .visually-hidden button { display: none !important; }
+    
+It is possible to apply the `.visually-hidden` class to content that contains nativily focusable elements (such as `a`, `button`, `input`, etc). It is important to still show these elements visually when they recieve focus, otherwise a keyboard-only users might not know what has focus. CSS for this might look something like:
+
+    .visually-hidden a:focus, .visually-hidden input:focus, .visually-hidden button:focus { position:static; width:auto; height:auto;  }
 
 Consider adding these HTML classes and CSS rules to your base stylesheet and use them when applicable.
 
-[^bereast]: [Screen readers sometimes ignore display:none](http://www.456bereastreet.com/archive/200711/screen_readers_sometimes_ignore_displaynone/) 456 Berea Street (November 7, 2007)
-[^clip]: [Clip Your Hidden Content For Better Accessibility](http://web.archive.org/web/20160616144545/https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html) Yahoo Accessibility Blog  (October 23, 2012)
+## Alternatives to `display: none`
+
+The `aria-hidden="true"` HTML attribute is the logical oposite of the `.visually-hidden` class. It hides content from assistive technology, but not visually. This can be helpful in cases where there are visual cues that screen readers do not need to speak, such as icons (although you should provide some form of alternative text for icons).
+
+There may be cases where you want to use `aria-hidden` and also visually hide the content. This can be accomplished with some CSS like:
+
+    .my-compnent[aria-hidden="true"] { 
+        display: none; 
+    }
+    
+ Another way to hide content both visually and from assistive technology is the [HTML `hidden` attribute](https://html.spec.whatwg.org/multipage/interaction.html#the-hidden-attribute). To support older browsers like IE9, you might want to add the following css to your pages[^hidden]:
+ 
+     *[hidden] { display: none; }
```

Comment:
No need for the `*` selector here; `[hidden]` will already select all elements with the `hidden` attribute.